### PR TITLE
frontend: convert pod pages to TypeScript

### PR DESCRIPTION
**Pull request by spadgett**
/assign @TheRealJon @alecmerdler

### CODE REVIEW

File: frontend/public/components/overview/index.tsx [hunk 1]
```diff
@@ -1086,8 +1087,8 @@ type FirehoseItem = {
   [key: string]: any;
 };
 
-type FirehoseList = {
-  data?: K8sResourceKind[];
+type FirehoseList<T extends K8sResourceKind> = {
```
Comment (alecmerdler) [inline frontend/public/components/overview/index.tsx]: Nice!

**Comment by spadgett** [comment]
/retest

```
2019/04/16 17:09:02 Container test in pod e2e-aws completed successfully
{"component":"entrypoint","level":"error","msg":"Process did not finish before 4h0m0s timeout","time":"2019-04-16T20:01:08Z"}
2019/04/16 20:01:08 error: Process interrupted with signal interrupt, exiting in 2s ...
2019/04/16 20:01:08 cleanup: Deleting release pod release-latest
2019/04/16 20:01:08 cleanup: Deleting template e2e-aws
{"component":"entrypoint","level":"error","msg":"Process gracefully exited before 15s grace period","time":"2019-04-16T20:01:10Z"}
```

**Pull request merged**